modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/utils/JavaParserUtils.java

```diff
@@ -329,21 +326,26 @@ private static MuleVersion calculateOutputMinMuleVersion(Type outputType) {
     return calculatedMMV;
   }
 
-  private static MuleVersion calculateFieldMinMuleVersion(ExtensionParameter field) {
-    MuleVersion calculatedMMV = getMinMuleVersionFromAnnotations(field).orElse(FIRST_MULE_VERSION);
+  private static MuleVersion calculateFieldMinMuleVersion(ExtensionParameter field, Set<String> seenTypes) {
     Type parameterType = field.getType();
+    if (seenTypes.contains(parameterType.getTypeName())) {
+      return FIRST_MULE_VERSION;
```

Comment:
the current type has not finished resolving (think of the recursive stack) so we can't return that; thus we return the first version to cut the recursion. The first call will analyse all other fields (the ones that are not recursive), so no information is missed by doing this.

Reply:
and what if they are all recursive? or a single one that is recursive? Also, if the set is just for recursion control, then rename the arg to reflect that, otherwise future devs  could see this as just a cache.

Reply:
works for both cases. My point was that the first call will analyse the class completely, so the second call (the recursive one) returning the first mule version does not cause loss of information.
Ok, renaming the arg.